rio_tiler/models.py

```diff
@@ -75,3 +82,111 @@ class Metadata(Info):
+    crs: Optional[CRS] = attr.ib(default=None)
+
+    def __iter__(self):
+        """Allow for variable expansion (``arr, mask = ImageData``)"""
+        for i in (self.data, self.mask):
+            yield i
+
+    @classmethod
+    def create_from_list(cls, data: Sequence["ImageData"]):
+        """Create ImageData from a sequence of ImageData objects."""
+        arr = numpy.concatenate([img.data for img in data])
+        mask = numpy.all([img.mask for img in data], axis=0).astype(numpy.uint8) * 255
+        assets = [img.assets[0] for img in data if img.assets]
+
+        bounds_values = [img.bounds for img in data if img.bounds]
+        bounds = list(set(bounds_values))[0] if bounds_values else None
+
+        crs_values = [img.crs for img in data if img.crs]
+        crs = list(set(crs_values))[0] if crs_values else None
+
+        return cls(arr, mask, assets=assets, crs=crs, bounds=bounds)
+
+    def as_masked(self) -> numpy.ma.MaskedArray:
+        """return a numpy masked array."""
+        data = numpy.ma.array(self.data)
+        data.mask = self.mask == 0
+        return data
+
+    def data_as_image(self) -> numpy.ndarray:
+        """Return the data array reshaped into an image processing/visualization software friendly order.
+
+        (bands, rows, columns) -> (rows, columns, bands).
+
+        """
+        return reshape_as_image(self.data)
+
+    @property
+    def width(self) -> int:
+        """Width of the data array."""
+        return self.data.shape[2]
+
+    @property
+    def height(self) -> int:
+        """Height of the data array."""
+        return self.data.shape[1]
+
+    @property
+    def count(self) -> int:
+        """Number of band."""
+        return self.data.shape[0]
+
+    @property
+    def transform(self):
+        """Returns the affine transform."""
+        return (
+            from_bounds(*self.bounds, self.width, self.height)
+            if self.bounds
+            else Affine.scale(self.width, -self.height)
+        )
+
+    def post_process(
+        self,
+        in_range: Optional[Tuple[NumType, NumType]] = None,
+        out_range: Optional[Tuple[NumType, NumType]] = None,
+        color_formula: Optional[str] = None,
+    ):
+        """Post-process image data."""
+        if in_range:
+            rescale_arr = tuple(_chunks(in_range, 2))
+            if len(rescale_arr) != self.count:
+                rescale_arr = ((rescale_arr[0]),) * self.count
+
+            for bdx in range(self.count):
+                self.data[bdx] = numpy.where(
+                    self.mask,
+                    linear_rescale(
+                        self.data[bdx], in_range=rescale_arr[bdx], out_range=out_range
+                    ),
+                    0,
+                )
+
+        if color_formula:
+            self.data[self.data < 0] = 0
+            for ops in parse_operations(color_formula):
+                self.data = scale_dtype(ops(to_math_type(self.data)), numpy.uint8)
```

Comment:
```suggestion
        inplace: bool = False,
    ) -> 'ImageData':
        """Post-process image data."""
        data = self.data if inplace else self.data.copy()
        mask = self.mask if inplace else self.mask.copy()

        if in_range:
            rescale_arr = tuple(_chunks(in_range, 2))
            if len(rescale_arr) != self.count:
                rescale_arr = ((rescale_arr[0]),) * self.count

            for bdx in range(self.count):
                data[bdx] = numpy.where(
                    self.mask,
                    linear_rescale(
                        data[bdx], in_range=rescale_arr[bdx], out_range=out_range
                    ),
                    0,
                )

        if color_formula:
            data[data < 0] = 0
            for ops in parse_operations(color_formula):
                data = scale_dtype(ops(to_math_type(data)), numpy.uint8)

        return self if inplace else ImageData(data, mask, crs=self.crs, bounds=self.bounds, assets=self.assets)
```

Reply:
I think it makes sense to add the `inplace` method to let the user decide if they want to overwrite the object or create a new one!